### ISSUE TITLE
Default value for `complexity` (20) is not used when extending `eslint:recommended`

### ISSUE THREAD

**Issue by Mardak** [eslint/eslint#6021]
Per @platinumazure, split from https://github.com/eslint/eslint/issues/4808#issuecomment-215940482

**What version of ESLint are you using?**
2.9.0

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
`complexity: 2`

**What did you do? Please include the actual source code causing the issue.**
```js
(function() { 1 || 2 || 3 || 4 || 5 || 6 || 7 || 8 || 9 || 10 || 11 || 12 })
```

**What did you expect to happen?**
no error

**What actually happened? Please include the actual, raw output from ESLint.**
`1:2   error  Function 'anonymous' has a complexity of 12              complexity`

`eslint:recommended` sets a default of 11: `"complexity": ["off", 11]`
https://github.com/eslint/eslint/blob/master/conf/eslint.json#L141

Docs say it's 20: `it will warn when the cyclomatic complexity crosses the configured threshold (default is `20`)`
https://github.com/eslint/eslint/blame/master/docs/rules/complexity.md#L19

The rule is implemented with 20: `THRESHOLD = 20`
https://github.com/eslint/eslint/blob/master/lib/rules/complexity.js#L49

Probably safest to just remove the `11` from `eslint.json` similar to what was done for #4809 with most of the other rules changed in https://github.com/eslint/eslint/commit/e2f2b6682e966ea9c20a18d5e554050d43db6271

If that's removed, the only rule with a default set in the file would be `object-curly-spacing`.

**Comment by kaicataldo** [comment]
I labeled this as core because it didn't seem to fit well into any other category. I think this change makes sense, though it is a breaking change.

**Comment by mysticatea** [comment]
@IanVS Had keeping `complexity: ["off", 11]` in #4809 been intentional?

**Comment by IanVS** [comment]
Nope, looks like I just missed that one.

**Comment by nzakas** [comment]
@ianvs why did you label this as breaking?

**Comment by kaicataldo** [comment]
@nzakas I also thought it would be a breaking change, since the behavior at the consumer-level would be different. Since the number will increase, it will stop warning in places it currently would warn. Maybe I'm missing something, though.

**Comment by IanVS** [comment]
@gyandeeps labeled the similar changes in #4809 as breaking, so I assumed this would be for the same reason.

**Comment by nzakas** [comment]
Oh I see, it's actually a change to `eslint:recommended`, I misread. Breaking is correct.
